jwst/resample/resample_step.py

```diff
@@ -116,8 +116,16 @@ def process(self, input):
             util.update_s_region_imaging(model)
             model.meta.asn.pool_name = input_models.asn_pool_name
             model.meta.asn.table_name = input_models.asn_table_name
-            model.meta.resample.pixel_scale_ratio = self.pixel_scale_ratio
+
+            # if pixel_scale exists, it will override pixel_scale_ratio.
+            # calculate the actual value of pixel_scale_ratio based on pixel_scale
+            # because source_catalog uses this value from the header.
+            if not self.pixel_scale:
+                model.meta.resample.pixel_scale_ratio = self.pixel_scale_ratio
+            else:
+                model.meta.resample.pixel_scale_ratio = (self.pixel_scale ** 2) / model.meta.photometry.pixelarea_arcsecsq
```

Comment:
So this is calculating the ratio of the *areas* of output vs. input. Is that the way `pixel_scale_ratio` is supposed to be used, as opposed to the linear ratio? It could be, I'm just not sure.

Reply:
Oh, and the computation is done using `pixelarea_arcsecsq`; is `pixel_scale` in units of arcsec at this point in the code or is it in degrees? If the latter, the difference in units needs to be taken into account.

Reply:
its in degrees at this point so ive fixed that. still thinking about how the ratio should be calculated

Reply:
I don't see how line 128 has been fixed to take into account the fact that `pixel_scale` is in units of degrees at this point in the code. It's still dividing `pixel_scale` by `pixelarea_arcsecsq`, the latter of which is in units of arcsec.

Reply:
Oh wait, maybe I see it now. `pixel_scale` is the original value provided by the user, which is in units of arcsec, but `pscale`, which is used in the actual drizzling code, is the one that's in units of degrees. OK, never mind. And `pixel_scale_ratio` is supposed to be in the sense of output / input (e.g. a value of 0.5 mean the output pixel size will be one-half that of the input), so it appears that the order of the calculation is also correct (if the output `pixel_scale` is smaller than the original, then we'll get a number < 1.